src/analysis.rs

```diff
@@ -100,13 +100,11 @@ impl Cost {
     }
 
     /// Return whether the cost is within the budget of
-    /// the given transaction input.
+    /// the given script witness of a transaction input.
     #[cfg(feature = "elements")]
-    pub fn is_budget_valid(&self, txin: &elements::TxIn) -> bool {
+    pub fn is_budget_valid(&self, script_witness: &Vec<Vec<u8>>) -> bool {
```

Comment:
In 11c65439e77b18d4255e5e756d307e735b284576

Even though we expect the user to just pass `txin.witness.script_witness`, we should take a `&[Vec<u8>]` rather than a `&Vec<Vec<u8>>`.

Reply:
[consensus_encode](https://docs.rs/bitcoin/latest/bitcoin/consensus/encode/trait.Encodable.html#tymethod.consensus_encode) is not implemented for `&[Vec<u8>]`. This is why clippy didn't show a warning, which I was wondering about. I will update the commit message.

I could combine the encodings of each `Vec<u8>`, but this requires knowledge about the encoding.

Reply:
Ah, I see.

In that case we should update the doccomment of the function.